azkaban-common/src/main/java/azkaban/user/UserManager.java

```diff
@@ -49,4 +49,12 @@ public User getUser(String username, String password)
   public Role getRole(String roleName);
 
   public boolean validateProxyUser(String proxyUser, User realUser);
+
+  /**
+   * @param username e.g. user alias
+   * @param groupName e.g. name of hadoop headless group / LDAP group
+   * @return Returns true if the user belongs to a group. This is used when verifying user
+   * permission by checking its group membership
+   */
+  public boolean validateUserGroupMembership(String username, String groupName);
```

Comment:
How is this new interface different from existing `validateProxyUser` method? If I understand correctly the `proxyUser` parameter is essentially a group and the method should verify if `realUser` belongs to it.

Reply:
I think validateProxyUser only checks if the realUser is part of proxyUser or one of realUser's headless group has admin permission. So it is not exactly what this new interface is for.